src/cli/cli_tcp.cpp

```diff
@@ -74,13 +98,62 @@ template <> otError TcpExample::Process<Cmd("init")>(Arg aArgs[])
     }
     else
     {
-        if (aArgs[0] == "circular")
+        if (aArgs[0] == "tls")
         {
             mUseCircularSendBuffer = true;
+            mUseTLS                = true;
+
+            // mbedtls_debug_set_threshold(0);
+
+            mbedtls_ctr_drbg_init(&mCtrDrbg);
+            mbedtls_x509_crt_init(&mSrvCert);
+            mbedtls_pk_init(&mPKey);
+
+            mbedtls_ssl_init(&mSSLContext);
+            mbedtls_ssl_config_init(&mSSLConfig);
+            mbedtls_ssl_conf_rng(&mSSLConfig, mbedtls_ctr_drbg_random, &mCtrDrbg);
+            // mbedtls_ssl_conf_dbg(&mSSLConfig, MbedTlsDebugOutput, this);
+            mbedtls_ssl_conf_authmode(&mSSLConfig, MBEDTLS_SSL_VERIFY_NONE);
+            mbedtls_entropy_init(&mEntropy);
+
+            int rv = mbedtls_pk_parse_key(&mPKey, reinterpret_cast<const unsigned char *>(sSrvKey), sSrvKeyLength,
+                                          nullptr, 0);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_pk_parse_key returned %d", rv);
+            }
+
+            rv = mbedtls_ctr_drbg_seed(&mCtrDrbg, mbedtls_entropy_function, nullptr, nullptr, 0);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_ctr_drbg_seed returned %d", rv);
+            }
+
+            rv = mbedtls_x509_crt_parse(&mSrvCert, reinterpret_cast<const unsigned char *>(sSrvPem), sSrvPemLength);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_x509_crt_parse (1) returned %d", rv);
+            }
+            rv = mbedtls_x509_crt_parse(&mSrvCert, reinterpret_cast<const unsigned char *>(sCasPem), sCasPemLength);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_x509_crt_parse (2) returned %d", rv);
+            }
+            rv = mbedtls_ssl_setup(&mSSLContext, &mSSLConfig);
+            if (rv != 0)
+            {
+                OutputLine("mbedtls_ssl_setup returned %d", rv);
```

Comment:
I hit this error when using Mbed TLS 3.x, fixed it using a similar approach as in https://github.com/openthread/openthread/blob/main/src/core/meshcop/dtls.cpp#L290-L296

Reply:
I included similar code (`#if (MBEDTLS_VERSION_NUMBER >= 0x03020000)`) as well here --- can you check if the fix I implemented is the same as the one that worked for you earlier?